submissions/xT0PERx/OOP exercise/index.js

```diff
@@ -0,0 +1,82 @@
+class Mammal {
+  constructor(name, gender, friend, saying, species, legs) {
+    this.name = name;
+    this.gender = gender;
+    this.friend = friend;
+    this.saying = saying;
+    this.species = species;
+    this.legs = legs;
+  }
+  toString() {
+    return [
+      this.name,
+      this.gender,
+      this.friend,
+      this.saying,
+      this.species,
+      this.legs,
+    ].join(";");
+  }
+}
+
+class Human extends Mammal {
+  constructor(
+    name,
+    gender,
+    friend,
+    saying,
+    species = "human",
+    legs = 2,
+    hands = 2
+  ) {
+    super(name, gender, friend, saying, species, legs);
+    this.hands = hands;
+  }
+  toString() {
+    return [super.toString(), this.hands].join(";");
+  }
+}
+
+class Cat extends Mammal {
+  constructor(name, gender, saying, friend, species = "cat", legs = 4) {
+    super(name, gender, friend, saying, species, legs);
+  }
+  toString() {
+    return super.toString();
+  }
+}
+class Dog extends Mammal {
+  constructor(name, gender, saying, species = "dog", friend, legs = 4) {
+    super(name, gender, friend, saying, species);
+    this.legs = legs;
```

Comment:
Is there a good reason for `Dog` to handle legs in a way different from `Cat`?

Reply:
Again my inattention :(

Reply:
Please take learning on learning process. This type of excuse is OK while you're learning but will be bad excuse in professional work.